rts/motoko-rts/src/idl.rs

```diff
@@ -130,6 +156,12 @@ unsafe fn parse_idl_header<M: Memory>(
     typtbl_size_out: *mut u32,
     main_types_out: *mut *mut u8,
 ) {
+    let mode = if extended {
```

Comment:
I guess this means we never validate the type table used for MemoryCompatibility. I wonder if we should (in future, not now)

Reply:
This is a good question. My assumption is that we trust the compiler and previous versions. Adding a check could be an additional safety guard to detect invalid or incompatible persistent memory (of previous compiler version), if we see possibility of such cases.